app/src/main/java/com/github/palFinderTeam/palfinder/meetups/FirebaseMeetUpService.kt

```diff
@@ -0,0 +1,97 @@
+package com.github.palFinderTeam.palfinder.meetups
+
+import com.firebase.geofire.GeoFireUtils
+import com.firebase.geofire.GeoLocation
+import com.github.palFinderTeam.palfinder.meetups.MeetUp.Companion.toMeetUp
+import com.github.palFinderTeam.palfinder.utils.Location
+import com.google.android.gms.tasks.Tasks
+import com.google.firebase.firestore.DocumentSnapshot
+import com.google.firebase.firestore.FirebaseFirestore
+import kotlinx.coroutines.tasks.await
+
+
+/**
+ * Object containing methods to query the database about MeetUps.
+ */
+object FirebaseMeetUpService : MeetUpRepository {
+
+    private const val MEETUP_COLL = "meetups"
+
+    override suspend fun getMeetUpData(meetUpId: String): MeetUp? {
+        val db = FirebaseFirestore.getInstance()
+
+        return try {
+            db.collection(MEETUP_COLL)
+                .document(meetUpId).get().await().toMeetUp()
+        } catch (e: Exception) {
+            null
+        }
+    }
+
+    override suspend fun createMeetUp(newMeetUp: MeetUp): String? {
+        val db = FirebaseFirestore.getInstance()
+
+        return try {
+            db.collection(MEETUP_COLL).add(newMeetUp.toFirestoreData()).await().id
+        } catch (e: Exception) {
+            null
+        }
+    }
+
+    override suspend fun editMeetUp(meetUpId: String, field: String, value: Any): String? {
```

Comment:
Why not but for something that simple, the complexity addition of having nullable parameters would make the api unclear, while having two overloading make it a bit simpler to use (actually only one is used, the one were we upload the whole meetup so I could remove the other, I just thought it could be nice to be able to only edit one field at the time, with user addtion for instance).